src/MerkleGen.sol

```diff
@@ -272,4 +272,134 @@ library MerkleGen {
 
         return (proof_hashes, proof_flag_bits_bool, root);
     }
+
+    function gen(bytes32[] memory leaves, uint256 leafIndex) internal pure returns (bytes32[] memory, bytes32) {
```

Comment:
The purpose of the library is for people to be able to generate merkle tree proofs, rather than the prove function. It'd be better to have the `gen` visibility as `public`.

We should also clarify the difference between `gen()` for and `gen()` for multi proof vs single. Let's update each of the names to make it clearer. e.g. `generateMultiproof()` and `generateSingleProof()` or something along those lines.

Reply:
Addressed.